src/back.jl

```diff
@@ -44,19 +47,39 @@ back_(::Call{Missing}, Δ, once) = error("`back!` was already used")
 accum!(x, Δ) = x .+ Δ
 accum!(x::AbstractArray, Δ) = (x .+= Δ)
 
+# function back(x::Tracked, Δ, once)
+#   x.isleaf && (x.grad = accum!(x.grad, Δ); return)
+#   ref = x.ref -= 1
+#   grad = if isdefined(x, :grad)
+#     x.grad = accum!(x.grad, Δ)
+#   elseif ref > 0
+#     x.grad = Δ
+#   else
+#     Δ
+#   end
+#   if ref == 0
+#     back_(x.f, grad, once)
+#     once && !x.isleaf && (x.f = Call(missing, ()))
+#   end
+#   return
+# end
+
+
 function back(x::Tracked, Δ, once)
-  x.isleaf && (x.grad = accum!(x.grad, Δ); return)
-  ref = x.ref -= 1
-  grad = if isdefined(x, :grad)
-    x.grad = accum!(x.grad, Δ)
-  elseif ref > 0
-    x.grad = Δ
-  else
-    Δ
-  end
-  if ref == 0
-    back_(x.f, grad, once)
-    once && !x.isleaf && (x.f = Call(missing, ()))
+  if !x.isleaf  # If x is not a leaf node
+    ref = getproperty(x, :ref, 0)  # Get the ref count of x, default to 0 if not available
+    grad = getproperty(x, :grad, nothing)  # Get the gradient of x, default to nothing if not available
```

Comment:
This syntax for `getproperty` is not supported. Type `?getproperty` at the REPL to see how to properly use this function.

Reply:
Okay, Thanks.